==22318== Memcheck, a memory error detector
==22318== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==22318== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==22318== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global
==22318== 
==22318== Conditional jump or move depends on uninitialised value(s)
==22318==    at 0x68E95D: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x68445C: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x6836D1: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29CE6C: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x268758: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57F2D4: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x25EDB1: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x260994: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x25F272: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x730B12: {{closure}} (rt.rs:52)
==22318==    by 0x730B12: do_call<closure-0,i32> (panicking.rs:348)
==22318==    by 0x730B12: try<i32,closure-0> (panicking.rs:325)
==22318==    by 0x730B12: catch_unwind<closure-0,i32> (panic.rs:394)
==22318==    by 0x730B12: std::rt::lang_start_internal (rt.rs:51)
==22318==    by 0x2622F1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==  Uninitialised value was created by a stack allocation
==22318==    at 0x29CD80: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 
==22318== HEAP SUMMARY:
==22318==     in use at exit: 30,384 bytes in 91 blocks
==22318==   total heap usage: 832 allocs, 741 frees, 143,037 bytes allocated
==22318== 
==22318== 16 bytes in 1 blocks are still reachable in loss record 1 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x6C28D8: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x6C7116: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x6CF260: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x6C1E25: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x58538D: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57F0B0: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2614CD: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x25F272: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x730B12: {{closure}} (rt.rs:52)
==22318==    by 0x730B12: do_call<closure-0,i32> (panicking.rs:348)
==22318==    by 0x730B12: try<i32,closure-0> (panicking.rs:325)
==22318==    by 0x730B12: catch_unwind<closure-0,i32> (panic.rs:394)
==22318==    by 0x730B12: std::rt::lang_start_internal (rt.rs:51)
==22318==    by 0x2622F1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 32 bytes in 1 blocks are still reachable in loss record 2 of 31
==22318==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==22318==    by 0x4E3F165: dlsym (dlsym.c:70)
==22318==    by 0x73F0E3: fetch (weak.rs:61)
==22318==    by 0x73F0E3: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==22318==    by 0x73F0E3: min_stack_size (thread.rs:449)
==22318==    by 0x73F0E3: std::sys::unix::thread::Thread::new (thread.rs:49)
==22318==    by 0x2E0666: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 40 bytes in 1 blocks are still reachable in loss record 3 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x6CE8BB: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x6C28C8: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x6C7116: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x6CF260: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x6C1E25: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x58538D: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57F0B0: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2614CD: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x25F272: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x730B12: {{closure}} (rt.rs:52)
==22318==    by 0x730B12: do_call<closure-0,i32> (panicking.rs:348)
==22318==    by 0x730B12: try<i32,closure-0> (panicking.rs:325)
==22318==    by 0x730B12: catch_unwind<closure-0,i32> (panic.rs:394)
==22318==    by 0x730B12: std::rt::lang_start_internal (rt.rs:51)
==22318== 
==22318== 64 bytes in 4 blocks are still reachable in loss record 4 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E991C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EA907: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3DA6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 64 bytes in 4 blocks are still reachable in loss record 5 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x73F03C: alloc (alloc.rs:80)
==22318==    by 0x73F03C: alloc (alloc.rs:174)
==22318==    by 0x73F03C: exchange_malloc (alloc.rs:277)
==22318==    by 0x73F03C: std::sys::unix::thread::Thread::new (thread.rs:44)
==22318==    by 0x2E0666: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 72 bytes in 1 blocks are still reachable in loss record 6 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x64619D: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x649E0B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x289C45: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x283F15: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57F68E: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57E003: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x260C79: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x25F272: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x730B12: {{closure}} (rt.rs:52)
==22318==    by 0x730B12: do_call<closure-0,i32> (panicking.rs:348)
==22318==    by 0x730B12: try<i32,closure-0> (panicking.rs:325)
==22318==    by 0x730B12: catch_unwind<closure-0,i32> (panic.rs:394)
==22318==    by 0x730B12: std::rt::lang_start_internal (rt.rs:51)
==22318==    by 0x2622F1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 128 bytes in 4 blocks are still reachable in loss record 7 of 31
==22318==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==22318==    by 0x70B66E: register_dtor (fast_thread_local.rs:33)
==22318==    by 0x70B66E: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==22318==    by 0x70B66E: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==22318==    by 0x72DE69: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==22318==    by 0x72DE69: __getit (local.rs:175)
==22318==    by 0x72DE69: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==22318==    by 0x72DE69: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==22318==    by 0x72DE69: std::sys_common::thread_info::set (thread_info.rs:40)
==22318==    by 0x2E07EC: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x73F459: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22318==    by 0x73F459: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22318==    by 0x73F459: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22318==    by 0x52516DA: start_thread (pthread_create.c:463)
==22318==    by 0x57A271E: clone (clone.S:95)
==22318== 
==22318== 128 bytes in 4 blocks are still reachable in loss record 8 of 31
==22318==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==22318==    by 0x2F20DD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E9AC2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E1586: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5219: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E32EA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5F30: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E082E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x73F459: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22318==    by 0x73F459: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22318==    by 0x73F459: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22318==    by 0x52516DA: start_thread (pthread_create.c:463)
==22318==    by 0x57A271E: clone (clone.S:95)
==22318== 
==22318== 160 bytes in 4 blocks are still reachable in loss record 9 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E7A4C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E4E1B: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EAC88: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5D50: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3E4A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 160 bytes in 4 blocks are still reachable in loss record 10 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E7A4C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E4E46: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EAC88: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5D50: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3E4A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 160 bytes in 4 blocks are still reachable in loss record 11 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E7A4C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EAAEB: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3EC0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 160 bytes in 4 blocks are still reachable in loss record 12 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x70CA0E: alloc (alloc.rs:80)
==22318==    by 0x70CA0E: alloc (alloc.rs:174)
==22318==    by 0x70CA0E: exchange_malloc (alloc.rs:277)
==22318==    by 0x70CA0E: new<()> (mutex.rs:223)
==22318==    by 0x70CA0E: std::thread::Thread::new (mod.rs:1149)
==22318==    by 0x2E0551: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 160 bytes in 4 blocks are still reachable in loss record 13 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E0573: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 192 bytes in 4 blocks are still reachable in loss record 14 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x729B06: alloc (alloc.rs:80)
==22318==    by 0x729B06: alloc (alloc.rs:174)
==22318==    by 0x729B06: exchange_malloc (alloc.rs:277)
==22318==    by 0x729B06: std::sync::condvar::Condvar::new (condvar.rs:130)
==22318==    by 0x2E4E26: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EAC88: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5D50: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3E4A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 192 bytes in 4 blocks are still reachable in loss record 15 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x729B06: alloc (alloc.rs:80)
==22318==    by 0x729B06: alloc (alloc.rs:174)
==22318==    by 0x729B06: exchange_malloc (alloc.rs:277)
==22318==    by 0x729B06: std::sync::condvar::Condvar::new (condvar.rs:130)
==22318==    by 0x2E4E51: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EAC88: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5D50: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3E4A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 192 bytes in 4 blocks are still reachable in loss record 16 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x729B06: alloc (alloc.rs:80)
==22318==    by 0x729B06: alloc (alloc.rs:174)
==22318==    by 0x729B06: exchange_malloc (alloc.rs:277)
==22318==    by 0x729B06: std::sync::condvar::Condvar::new (condvar.rs:130)
==22318==    by 0x2EAAF1: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3EC0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 192 bytes in 4 blocks are still reachable in loss record 17 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x70CA72: alloc (alloc.rs:80)
==22318==    by 0x70CA72: alloc (alloc.rs:174)
==22318==    by 0x70CA72: exchange_malloc (alloc.rs:277)
==22318==    by 0x70CA72: new (condvar.rs:130)
==22318==    by 0x70CA72: std::thread::Thread::new (mod.rs:1150)
==22318==    by 0x2E0551: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 320 bytes in 4 blocks are still reachable in loss record 18 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x70CAC1: alloc (alloc.rs:80)
==22318==    by 0x70CAC1: alloc (alloc.rs:174)
==22318==    by 0x70CAC1: exchange_malloc (alloc.rs:277)
==22318==    by 0x70CAC1: new<std::thread::Inner> (sync.rs:324)
==22318==    by 0x70CAC1: std::thread::Thread::new (mod.rs:1145)
==22318==    by 0x2E0551: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 384 bytes in 1 blocks are still reachable in loss record 19 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E8F6C: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5D0C: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3E4A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 416 bytes in 4 blocks are still reachable in loss record 20 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E0608: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 512 bytes in 1 blocks are still reachable in loss record 21 of 31
==22318==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x72F579: aligned_malloc (alloc.rs:95)
==22318==    by 0x72F579: alloc (alloc.rs:22)
==22318==    by 0x72F579: __rdl_alloc (alloc.rs:304)
==22318==    by 0x2E9186: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3E7A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 640 bytes in 1 blocks are still reachable in loss record 22 of 31
==22318==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x72F579: aligned_malloc (alloc.rs:95)
==22318==    by 0x72F579: alloc (alloc.rs:22)
==22318==    by 0x72F579: __rdl_alloc (alloc.rs:304)
==22318==    by 0x2E40B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 640 bytes in 1 blocks are still reachable in loss record 23 of 31
==22318==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x72F579: aligned_malloc (alloc.rs:95)
==22318==    by 0x72F579: alloc (alloc.rs:22)
==22318==    by 0x72F579: __rdl_alloc (alloc.rs:304)
==22318==    by 0x2F24E2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2F217D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E9AC2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E1586: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5219: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E32EA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5F30: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E082E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 808 bytes in 1 blocks are still reachable in loss record 24 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x6516FF: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x64A211: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x289C45: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x283F15: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57F68E: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x57E003: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x260C79: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x25F272: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x730B12: {{closure}} (rt.rs:52)
==22318==    by 0x730B12: do_call<closure-0,i32> (panicking.rs:348)
==22318==    by 0x730B12: try<i32,closure-0> (panicking.rs:325)
==22318==    by 0x730B12: catch_unwind<closure-0,i32> (panic.rs:394)
==22318==    by 0x730B12: std::rt::lang_start_internal (rt.rs:51)
==22318==    by 0x2622F1: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 1,152 bytes in 4 blocks are possibly lost in loss record 25 of 31
==22318==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==22318==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==22318==    by 0x5252227: allocate_stack (allocatestack.c:627)
==22318==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==22318==    by 0x73F18C: std::sys::unix::thread::Thread::new (thread.rs:66)
==22318==    by 0x2E0666: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3906: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E44CF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 1,520 bytes in 1 blocks are still reachable in loss record 26 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E3F12: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2A3321: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x287B88: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 1,536 bytes in 4 blocks are still reachable in loss record 27 of 31
==22318==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x72F579: aligned_malloc (alloc.rs:95)
==22318==    by 0x72F579: alloc (alloc.rs:22)
==22318==    by 0x72F579: __rdl_alloc (alloc.rs:304)
==22318==    by 0x2E99C0: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EA907: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3DA6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 2,008 bytes in 1 blocks are still reachable in loss record 28 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2F1C2E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2F24B2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2F217D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E9AC2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E1586: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5219: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E32EA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5F30: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E082E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x73F459: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22318==    by 0x73F459: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22318==    by 0x73F459: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22318== 
==22318== 4,096 bytes in 4 blocks are still reachable in loss record 29 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E9900: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2EA907: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E3DA6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E7655: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x72A1B9: std::sync::once::Once::call_inner (once.rs:416)
==22318==    by 0x2E3A85: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E6A34: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x28FE31: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2963A5: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x29A8A8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2B68E1: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318== 
==22318== 6,080 bytes in 4 blocks are still reachable in loss record 30 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2E2CB8: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5F30: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E082E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x73F459: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22318==    by 0x73F459: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22318==    by 0x73F459: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22318==    by 0x52516DA: start_thread (pthread_create.c:463)
==22318==    by 0x57A271E: clone (clone.S:95)
==22318== 
==22318== 8,160 bytes in 4 blocks are still reachable in loss record 31 of 31
==22318==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==22318==    by 0x2EF50F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2F2104: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E9AC2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E1586: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5219: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E32EA: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E5F30: std::sys_common::backtrace::__rust_begin_short_backtrace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x2E082E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==22318==    by 0x73F459: call_once<(),FnOnce<()>> (boxed.rs:1081)
==22318==    by 0x73F459: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==22318==    by 0x73F459: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==22318==    by 0x52516DA: start_thread (pthread_create.c:463)
==22318==    by 0x57A271E: clone (clone.S:95)
==22318== 
==22318== LEAK SUMMARY:
==22318==    definitely lost: 0 bytes in 0 blocks
==22318==    indirectly lost: 0 bytes in 0 blocks
==22318==      possibly lost: 1,152 bytes in 4 blocks
==22318==    still reachable: 29,232 bytes in 87 blocks
==22318==         suppressed: 0 bytes in 0 blocks
==22318== 
==22318== For counts of detected and suppressed errors, rerun with: -v
==22318== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)
